### PR TITLE
*: update google/btree to improve performance

### PR DESCRIPTION
**Pull request by hawkingrei**

### CODE REVIEW

File: internal/locate/sorted_btree_generic.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
+// Copyright 2021 TiKV Authors
```
Comment (sticnarf) [inline internal/locate/sorted_btree_generic.go:1]:
```suggestion
// Copyright 2022 TiKV Authors
```

File: internal/locate/sorted_btree.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
+// Copyright 2021 TiKV Authors
```
Comment (sticnarf) [inline internal/locate/sorted_btree.go:1]:
```suggestion
// Copyright 2022 TiKV Authors
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -35,15 +35,17 @@ jobs:
```
Comment (sticnarf) [inline .github/workflows/test.yml:35]: Could you change the go version above to [1.17, 1.18]? At this time, let's test it work under both 1.17 and 1.18.

File: internal/locate/region_cache.go [hunk 4]
```diff
@@ -371,7 +371,7 @@ type RegionCache struct {
 		sync.RWMutex                           // mutex protect cached region
 		regions        map[RegionVerID]*Region // cached regions are organized as regionVerID to region ref mapping
 		latestVersions map[uint64]RegionVerID  // cache the map from regionID to its latest RegionVerID
-		sorted         *btree.BTree            // cache regions are organized as sorted key to region ref mapping
+		sorted         *Sorted                 // cache regions are organized as sorted key to region ref mapping
```
Comment (sticnarf) [inline internal/locate/region_cache.go:374]: Let's call this type `SortedRegions`. It's a specialization of the generic `BTree[Region]`.

Comment (hawkingrei) [reply internal/locate/region_cache.go]: OK, I have changed it.

File: .github/workflows/test.yml (hunk 3 above)
Comment (hawkingrei) [reply .github/workflows/test.yml:38]: Done

**Review by sticnarf — APPROVED** [review]
LGTM

**Pull request merged**